src/persist-types/src/part.rs

```diff
@@ -0,0 +1,551 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A columnar representation of one blob's worth of data
+
+use std::any::Any;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use arrow2::array::{Array, NullArray, PrimitiveArray, StructArray};
+use arrow2::buffer::Buffer;
+use arrow2::chunk::Chunk;
+use arrow2::datatypes::{DataType as ArrowLogicalType, Field};
+use arrow2::io::parquet::write::Encoding;
+
+use crate::columnar::sealed::ColumnRef;
+use crate::columnar::{ColumnFormat, Data, DataType, Schema};
+
+/// A columnar representation of one blob's worth of data.
+#[derive(Debug, Default)]
+pub struct Part {
```

Comment:
dropped a note in Slack on this one, but it's not super clear to me what `Part` does versus an in-mem arrow representation